src/test/java/org/opensearch/knn/bwc/KNNBackwardsCompatibilityIT.java

```diff
@@ -194,4 +226,41 @@ private String getUri(ClusterType clusterType) {
                 throw new IllegalArgumentException("unknown cluster type: " + clusterType);
         }
     }
+
+   private double getkNNBWCRecallValue(String testIndex, String testField, int docCount, int dimensions, int queryCount, int k, boolean isStandard, SpaceType spaceType) throws Exception {
+       float[][] indexVectors = getIndexVectorsFromIndex(testIndex, testField, docCount, dimensions);
+       float[][] queryVectors = TestUtils.getQueryVectors(queryCount, dimensions, docCount, isStandard);
+       List<Set<String>> groundTruthValues = TestUtils.computeGroundTruthValues(indexVectors, queryVectors, spaceType, k);
+       List<List<String>> searchResults = getSearchResults(testIndex, testField, queryVectors, k);
+       return TestUtils.calculateRecallValue(searchResults, groundTruthValues, k);
+   }
+
+    private void addDocs(String testIndex, String testField, int dimensions, int docCount, boolean isStandard) throws Exception {
+        createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMapping(testField, dimensions));
+
+        updateClusterSettings(KNN_ALGO_PARAM_INDEX_THREAD_QTY, 2);
+        updateClusterSettings(KNN_MEMORY_CIRCUIT_BREAKER_ENABLED, true);
+
+        bulkAddKnnDocs(testIndex, testField, TestUtils.getIndexVectors(docCount, dimensions, isStandard), docCount);
+    }
+
+    private List<List<String>> getSearchResults(String testIndex, String testField, float[][] queryVectors, int k) throws IOException {
```

Comment:
We can't create another abstract class because again we can't extend 2 classes(abstract class and KNNRestTestCase). 

For now, as discussed offline will move getSearchResults to knnRestTestCase.